lib/ontologies_linked_data/models/mod/semantic_artefact_catalog.rb

```diff
@@ -0,0 +1,285 @@
+            def set_supported_schema
+                []
+            end
+            
+            def metrics_url
+                RDF::URI(LinkedData.settings.id_url_prefix).join('metrics')
+            end
+
+            def artefacts_url
+                RDF::URI(LinkedData.settings.id_url_prefix).join('artefacts')
+            end
+
+            def get_services
+                []
+            end
+
+            def records_url
+                RDF::URI(LinkedData.settings.id_url_prefix).join('records')
+            end
+
+            def distributions_url
+                RDF::URI(LinkedData.settings.id_url_prefix).join('distributions')
+            end
+            
+            def class_count
+                calculate_attr_from_metrics(:classes)
+            end
+            
+            def individuals_count
+                calculate_attr_from_metrics(:individuals)
+            end
+            
+            def propoerties_count
+                calculate_attr_from_metrics(:properties)
+            end
+            
+            def axioms_counts
+                calculate_attr_from_metrics(:numberOfAxioms)
+            end
+            
+            def object_properties_counts
+                calculate_attr_from_metrics(:numberOfObjectProperties)
+            end
+            
+            def data_properties_counts
+                calculate_attr_from_metrics(:numberOfDataProperties)
+            end
+            
+            def labels_counts
+                calculate_attr_from_metrics(:numberOfLabels)
+            end
+            
+            def deprecated_counts
+                calculate_attr_from_metrics(:numberOfDeprecated)
+            end
+            
+            def using_projects_counts
+                calculate_attr_from_metrics(:numberOfUsingProjects)
+            end
+            
+            def endorsements_counts
+                calculate_attr_from_metrics(:numberOfEnsorments)
+            end
+            
+            def mappings_counts
+                calculate_attr_from_metrics(:numberOfMappings)
+            end
+            
+            def users_counts
+                LinkedData::Models::User.all.count
+            end
+            
+            def agents_counts
+                LinkedData::Models::Agent.all.count
+            end
+
+            def self.valid_hash_code(inst, attr)
+                inst.bring(attr) if inst.bring?(attr)
+                str = inst.send(attr)
+        
+                return if (/^#(?:[0-9a-fA-F]{3}|[0-9a-fA-F]{6})$/ === str)
+                [:valid_hash_code,
+                 "Invalid hex color code: '#{str}'. Please provide a valid hex code in the format '#FFF' or '#FFFFFF'."]
+            end
+
+            private
+
+            def calculate_attr_from_metrics(attr)
+                attr_to_get = attr.to_sym
+                submissions = LinkedData::Models::OntologySubmission.where.include(OntologySubmission.goo_attrs_to_load([attr_to_get]))
```

Comment:
this code will be really slow, and sum all the submissions not only the latest.

why no just just do. `LinkedData::Models::Metric.where.include(...things_you_want...).all` it is faster, from that you need to filter the latest IDs only

Reply:
@imadbourouche can you open an issue about this,  I merged the PR forgot I said this.

Reply:
yes i'll fix this in another pull request before closing the main issue for catalog